src/Xamarin.CodeAnalysis/RequiresSuperAttributeCodeFixProvider.cs

```diff
@@ -0,0 +1,92 @@
+//
+// Copyright 2020 Xamarin Inc.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+
+
+namespace RequiresSuperAttribute
+{
+	[ExportCodeFixProvider (LanguageNames.CSharp, Name = nameof (RequiresSuperAnalyzerCodeFixProvider)), Shared]
+	public class RequiresSuperAnalyzerCodeFixProvider : CodeFixProvider
+	{
+		const string Title = "Add base method call";
+
+		public sealed override ImmutableArray<string> FixableDiagnosticIds {
+			get { return ImmutableArray.Create (RequiresSuperAttributeAnalyzer.DiagnosticId); }
+		}
+
+		public sealed override FixAllProvider GetFixAllProvider ()
+		{
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public sealed override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false);
+			var diagnostic = context.Diagnostics.First ();
+			var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+			// Find the method declaration identified by the diagnostic.
+			var methodDeclaration =
+			  root.FindToken (diagnosticSpan.Start).Parent.AncestorsAndSelf ()
+			  .OfType<MethodDeclarationSyntax> ().First ();
+
+			// Register a code action that will invoke the fix.
+			context.RegisterCodeFix (CodeAction.Create (Title, c => FixSuperAsync (context.Document, methodDeclaration, c), equivalenceKey: Title), diagnostic);
+		}
+
+		private async Task<Document> FixSuperAsync (Document document,
+		  MethodDeclarationSyntax methodDeclaration,
+		  CancellationToken cancellationToken)
+		{
+			//Find the node to fix.
+			var semanticModel = await document.GetSemanticModelAsync (cancellationToken);
+			var memberSymbol = semanticModel.GetDeclaredSymbol (methodDeclaration) as IMethodSymbol;
+
+			//Generate the new base.[methodName]([arguments]) block. If the method needs to return the base call,
+			//it generates return base.[methodName]([arguments])
+			// TODO: this might be hacky - (.Add(SyntaxFactory.Tab) adds an extra tab so that it's a tab (or SyntaxFactory.Tab) further than the leading
+			//      trivia of the method body block, but we should consider investigating if this behaves as expected when the IDE is set to spaces vs
+			//      tabs / different numbers of spaces per tab
+			var returnString = memberSymbol.ReturnsVoid ? "" : "return ";
+			var parametersString = string.Join (",", memberSymbol.Parameters.Select ((p) => p.Name).ToArray ()); //creates a string of the method's passed-in arguments separated by commas
+			var newLiteral = SyntaxFactory.ParseStatement (returnString + "base." + memberSymbol.Name + "(" + parametersString + ");")
```

Comment:
Could be nicer with:
```suggestion
			var newLiteral = SyntaxFactory.ParseStatement ("{returnString} base.{memberSymbol.Name} ({parametersString});")
```
Both result in string.Concat but one is more readable than the other.

Reply:
^ in that suggested change, I think there's meant to be a `$` prepending the string, so it's:
`$"{returnString} base.{memberSymbol.Name} ({parametersString});"`

Reply:
@xxaxdxcxx good catch